do-not-commit/insecure-deserialization.py

```diff
@@ -0,0 +1,49 @@
+# example from https://medium.com/gdg-vit/deserialization-attacks-d312fbe58e7d
+# flask_app.py
+import os
+import pickle
+from base64 import b64decode
+from base64 import b64encode
+from uuid import uuid1
+
+from flask import Flask
+from flask import make_response
+from flask import request
+
+# The User Class which assigns a random ID to each connection
+class UserID:
+    def __init__(self, uuid=None):
+        self.uuid = str(uuid1())
+
+    def __str__(self):
+        return self.uuid
+
+
+# The main Flask Backend
+app = Flask(__name__)
+
+
+@app.route("/", methods=["GET"])
+def index():
+    user_obj = request.cookies.get("uuid")
+    if user_obj == None:
+        msg = "Seems like you didn't have a cookie. No worries! I'll set one now!"
+        response = make_response(msg)
+        user_obj = UserID()
+        # ruleid:insecure-deserialization
+        response.set_cookie("uuid", b64encode(pickle.dumps(user_obj)))
+        return response
+    else:
+        # ruleid:insecure-deserialization
+        return "Hey there! {}!".format(pickle.loads(b64decode(user_obj)))
```

Comment:
Avoid using `pickle`, which is known to lead to code execution vulnerabilities.
When unpickling, the serialized data could be manipulated to run arbitrary code.
Instead, consider serializing the relevant data as JSON or a similar text-based
serialization format. ([serialization.pickle.avoid-pickle](https://semgrep.dev/r/serialization.pickle.avoid-pickle))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=serialization.pickle.avoid-pickle) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>